.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,56 @@
+name: Static Analysis
+on: [push, workflow_dispatch]
+jobs:
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the source code
+        uses: actions/checkout@v3
+
+      - name: Install & display rust toolchain
+        run: rustup show
+
+      - name: Check targets are installed correctly
+        run: rustup target list --installed
+
+      - name: Check fmt
+        run: cargo fmt -- --check
+
+      - name: Check Cargo.toml format
+        run: |
+          if taplo --version &> /dev/null; then
+            echo "taplo-cli is already installed"
+          else
+            cargo install taplo-cli
+          fi
+          taplo fmt --check
+
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the source code
+      uses: actions/checkout@v3
+
+    - name: Install & display rust toolchain
+      run: rustup show
+
+    - name: Check targets are installed correctly
+      run: rustup target list --installed
+
+    - name: Add Clippy
```

Comment:
Can you just modify the `rust-toolchain.toml` file?

components = [ "rustfmt", "clippy" ]